### PR TITLE
Do not shutdown FPGA

### PR DESCRIPTION
**Pull request by yashi**
This PR implements FPGA reconfiguration instead of power cycling in the state machine.

### CODE REVIEW

File: src/fpga.c [hunk 1]
```diff
@@ -138,7 +138,26 @@ static enum FpgaState trans_to_power_up(void)
         return FPGA_STATE_POWER_UP;
 }
 
-static enum FpgaState trans_to_ready(void)
+static enum FpgaState trans_to_ready_from_error()
```
Comment (sasataku) [inline src/fpga.c:141]: Is it better to define the void argument ?

File: src/fpga.c [hunk 2]
```diff
@@ -138,7 +138,26 @@ static enum FpgaState trans_to_power_up(void)
...
+        /* FPGAPWR_EN_DIR keep */
+        /* I2C_INT_SCL_DIR keep */
+        /* I2C_INT_SDA_DIR keep */
+        /* I2C_EXT_SCL_DIR keep */
+        /* I2C_EXT_SDA_DIR keep */
+
+        return FPGA_STATE_READY;
+}
+
+static enum FpgaState trans_to_ready()
```
Comment (sasataku) [inline src/fpga.c:160]: Is it better to define the void argument ?

File: src/main.c [hunk 3]
```diff
@@ -236,7 +236,8 @@ void main (void)
                         break;
 
                 case FPGA_STATE_ERROR:
-                        activate_fpga = false;
+                        activate_fpga = FPGA_SHUTDOWN;
+                        /* activate_fpga = FPGA_RECONFIGURE; */
```
Comment (sasataku) [inline src/main.c:240]: Is is intentional comment out ?

**Review by sasataku — CHANGES_REQUESTED** [review]

File: src/fpga.c (hunk 1 above)
Comment (yashi) [reply src/fpga.c]: done

File: src/main.c (hunk 3 above)
Comment (yashi) [reply src/main.c:240]: Yes it is.

File: src/fpga.c (hunk 2 above)
Comment (yashi) [reply src/fpga.c]: done

**Pull request merged**